tests/units/result_manager/test__init__.py

```diff
@@ -149,6 +171,95 @@ def test_add(
+        result_manager = ResultManager()
+        with pytest.raises(TypeError, match="Added test result 'test' must be a TestResult instance, got str."):
+            result_manager.add("test")  # type: ignore[arg-type]
+
+    def test_get_results(self, test_result_factory: Callable[[], TestResult]) -> None:
+        """Test ResultManager.get_results."""
+        result_manager = ResultManager()
+        assert result_manager.get_results() == []
+
+        success_result = test_result_factory()
+        success_result.result = "success"
+        success_result.categories = ["ospf"]
+        result_manager.add(success_result)
+
+        failure_result = test_result_factory()
+        failure_result.result = "failure"
+        failure_result.categories = ["bgp"]
+        result_manager.add(failure_result)
+
+        skipped_result = test_result_factory()
+        skipped_result.result = "skipped"
+        result_manager.add(skipped_result)
+
+        error_result = test_result_factory()
+        error_result.result = "error"
+        result_manager.add(error_result)
+
+        # Check for single status
+        success_results = result_manager.get_results(status="success")
+        assert len(success_results) == 1
+        assert success_results[0].result == "success"
+
+        # Check for multiple statuses
+        success_failure_results = result_manager.get_results(status={"success", "failure"})
+        assert len(success_failure_results) == 2
+        assert all(r.result in {"success", "failure"} for r in success_failure_results)
+
+        # Check all results
+        all_results = result_manager.get_results()
+        assert len(all_results) == 4
+        assert [r.result for r in all_results] == ["success", "failure", "skipped", "error"]
+
+        # Check all results with sort_by result
+        all_results = result_manager.get_results(sort_by=["result"])
+        assert len(all_results) == 4
+        assert [r.result for r in all_results] == ["error", "failure", "skipped", "success"]
+
+        # Check multiple statuses with sort_by categories
+        success_failure_results = result_manager.get_results(status={"success", "failure"}, sort_by=["categories"])
+        assert len(success_failure_results) == 2
+        assert success_failure_results[0] == failure_result
+        assert success_failure_results[1] == success_result
+
+        # Check all results with bad sort_by
+        with pytest.raises(
+            ValueError,
+            match=re.escape(
+                "Invalid sort_by fields: ['bad_field']. Accepted fields are: ['name', 'test', 'categories', 'description', 'result', 'messages', 'custom_field']",
+            ),
+        ):
+            all_results = result_manager.get_results(sort_by=["bad_field"])
+
+    def test_get_total_results(self, test_result_factory: Callable[[], TestResult]) -> None:
+        """Test ResultManager.get_total_results."""
+        result_manager = ResultManager()
```

Comment:
maybe we can build a new fixture doing what you want here?

Reply:
Yeah I think creating a ResultManager() with the 48 random tests could test this properly in a cleaner way (would also clean the other tests as well).

Reply:
Fixed in 73208a5aa2055f69a02daf2e7e18464a30a3983e